src/utils/udisks2_interfaces.py

```diff
@@ -0,0 +1,818 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, List, Tuple, Optional
+
+from sdbus import (
+    DbusInterfaceCommon,
+    DbusPropertyEmitsChangeFlag,
+    DbusUnprivilegedFlag,
+    dbus_method,
+    dbus_property,
+    DbusObjectManagerInterface,
+    SdBus,
+)
+from sdbus.utils import parse_get_managed_objects
+from sortedcollections import ValueSortedDict
+
+
+UDISKS2_SERVICE_NAME = "org.freedesktop.UDisks2"
+UDISKS2_BASE_PATH = "/org/freedesktop/UDisks2"
+UDISKS2_DRIVE_PATH = f"{UDISKS2_BASE_PATH}/drives"
+
+
+class GetDisksMixin(DbusInterfaceCommon):
+    def __init__(
+        self, service_name: str, object_path: str, bus: Optional[SdBus] = None
+    ):
+        super().__init__(service_name, object_path, bus)
+        self._object_manager = DbusObjectManagerInterface(
+            UDISKS2_SERVICE_NAME, UDISKS2_BASE_PATH, bus
+        )
+
+    def get_disks(self) -> dict[str, DiskResult]:
+        result = ValueSortedDict(lambda val: value.props["sort_key"])
+
+        for object_path, (iface, props) in parse_get_managed_objects(
+            (Disk, AtaDisk, NvmeDisk),
+            self._object_manager.get_managed_objects(),
+            on_unknown_interface="none",
+            on_unknown_member="ignore",
+        ).items():
+            if (
+                object_path.startswith(UDISKS2_DRIVE_PATH)  # This is a drive
+                and not props["optical"]  # This is not a CD player
+            ):
+                value = DiskResult(object_path, iface, props)
+                result[value.name] = value
+
+        return result
+
+
+class DiskResult:
+    def __init__(self, object_path: str, iface: type[Disk], props: dict[str, Any]):
+        self._object_path = object_path
+        self._iface = iface
+        self.name = object_path.removeprefix(f"{UDISKS2_DRIVE_PATH}/")
+        self.props = props
+
+
+"""
+The following interfaces were generated using the `python -m sdbus gen-from-file` command.
+See: https://python-sdbus.readthedocs.io/en/latest/code_generator.html
+Udisks2's dbus interface descriptors are available at:
+https://github.com/storaged-project/udisks/blob/2.10.x-branch/data/org.freedesktop.UDisks2.xml  
+"""
+
+
+class Udisks2Manager(
+    GetDisksMixin,
+    DbusInterfaceCommon,
+    interface_name="org.freedesktop.UDisks2.Manager",
+):
+
+    def __init__(self, bus: SdBus):
+        super().__init__(UDISKS2_SERVICE_NAME, f"{UDISKS2_BASE_PATH}/Manager", bus)
+
+    @dbus_method(
+        input_signature="s",
+        result_signature="(bs)",
+        flags=DbusUnprivilegedFlag,
+    )
+    def can_format(self, type: str) -> Tuple[bool, str]:
+        raise NotImplementedError
```

Comment:
Have you tried to delete all NotImplemented methods ?

Reply:
Those interfaces a automatically generated by the CLI as mentioned at the beginning of the file. The implementation is provided by `@dbus_method`. The `NotImplementerError` is just here to have a method content. It could be `pass` but that's what the CLI generates so I kept it.